node/node.go

```diff
@@ -115,6 +115,9 @@ type AlgorandFullNode struct {
 
 	log logging.Logger
 
+	// use syncMu for locking lastRoundTimestamp and hasSyncedSinceStartup
+	// syncMu added so OnNewBlock wouldn't be blocked by oldKeyDeletionThread during catchup
+	syncMu                deadlock.Mutex
 	lastRoundTimestamp    time.Time
 	hasSyncedSinceStartup bool
```

Comment:
When writing a variable description in go, try to always start with the variable name; go-vet would complain about that for public functions.